tests/contrib/anthropic/test_anthropic_llmobs.py

```diff
@@ -255,3 +258,401 @@ def test_image(self, anthropic, ddtrace_global_config, mock_llmobs_writer, mock_
                     tags={"ml_app": "<ml-app-name>"},
                 )
             )
+
+    @pytest.mark.skipif(ANTHROPIC_VERSION < (0, 27), reason="Anthropic Tools not available until 0.27.0, skipping.")
+    def test_tools_sync(self, anthropic, ddtrace_global_config, mock_llmobs_writer, mock_tracer, request_vcr):
+        """Ensure llmobs records are emitted for completion endpoints when configured and there is an stream input.
+
+        Also ensure the llmobs records have the correct tagging including trace/span ID for trace correlation.
+        """
+
+        llm = anthropic.Anthropic()
+        with request_vcr.use_cassette("anthropic_completion_tools.yaml"):
+            message = llm.messages.create(
+                model="claude-3-opus-20240229",
+                max_tokens=200,
+                messages=[{"role": "user", "content": "What is the weather in San Francisco, CA?"}],
+                tools=tools,
+            )
+            assert message is not None
+
+        traces = mock_tracer.pop_traces()
+        span_1 = traces[0][0]
+        mock_llmobs_writer.enqueue.assert_called_with(
+            _expected_llmobs_llm_span_event(
+                span_1,
+                model_name="claude-3-opus-20240229",
+                model_provider="anthropic",
+                input_messages=[{"content": "What is the weather in San Francisco, CA?", "role": "user"}],
+                output_messages=[
+                    {
+                        "content": "<thinking>\nThe get_weather tool is directly relevant for answering this "
+                        + "question about the weather in a specific location. \n\nThe get_weather tool "
+                        + 'requires a "location" parameter. The user has provided the location of "San '
+                        + 'Francisco, CA" in their question, so we have the necessary information to '
+                        + "make the API call.\n\nNo other tools are needed to answer this question. We "
+                        + "can proceed with calling the get_weather tool with the provided location."
+                        + "\n</thinking>",
+                        "role": "assistant",
```

Comment:
Can we declare this message once and reuse it? Seems unnecessary to hardcode the same 10-line message 4 times

Reply:
unfortunately the claude API responses vary between normal `.create`, `.stream`, `.create(stream=true)` calls, so we will never get the same message for all three cases. I don't think we can change this since we need the different response types for the cassettes

Reply:
I could probably change this in the future by using prompts that will return more basic messages (I'm thinking one word answers, but for streaming that may mean our testing isn't as strong since a basic message could be returned in one chunk). Anyhow that would probably put us out of our deadline timeframe

Reply:
No worries, let's just try to reuse the same message per test then. At the very least would save us a lot of lines

Reply:
fixed this, the only different one is the stream since the response text is different (but very similar)